sopel/modules/spellcheck.py

```diff
@@ -1,55 +1,151 @@
-Copyright © 2012, Lior Ramati
+spellcheck.py - Sopel spelling checker module
+Copyright © 2016, Alan Huang
+Copyright © 2019, dgw
 Licensed under the Eiffel Forum License 2.
-
 https://sopel.chat
-
-This module relies on pyenchant, on Fedora and Red Hat based system, it can be found in the package python-enchant
 """
 from __future__ import unicode_literals, absolute_import, print_function, division
+import aspell
+from sopel.module import commands, example, require_admin
+
+
+def setup(bot):
+    bot.memory['spellcheck_pending_adds'] = []
+
+
+def shutdown(bot):
+    try:
+        del bot.memory['spellcheck_pending_adds']
+    except KeyError:
+        pass
+
+
+@commands('scadd')
+@require_admin('I only trust admins to add words.')
+def add_command(bot, trigger):
+    """
+    Stage a word to be added to the bot's personal dictionary.
+    """
+    bot.memory['spellcheck_pending_adds'].append(trigger.group(2))
+    bot.say('Added "{0}". (To review pending words, use {1}scpending. '
+            'To commit changes, use {1}scsave.)'
+            .format(trigger.group(2), bot.config.core.help_prefix))
 
-try:
-    import enchant
-except ImportError:
-    enchant = None
-from sopel.module import commands, example
+
+@commands('scpending')
+def pending_command(bot, trigger):
+    """
+    List words that are waiting to be saved to the bot's personal dictionary.
+    """
+    bot.say('Ready to save: "{0}". (To remove a word before saving, use {1}scdel '
+            'word, or clear the list with {1}scclear.)'
+            .format('", "'.join(bot.memory['spellcheck_pending_adds']),
+                    bot.config.core.help_prefix))
 
 
-@commands('spellcheck', 'spell')
-@example('.spellcheck stuff')
-def spellcheck(bot, trigger):
+@commands('scdel')
+@require_admin('Only admins may cancel a word-list addition.')
+def del_command(bot, trigger):
     """
-    Says whether the given word is spelled correctly, and gives suggestions if
-    it's not.
+    Remove a word from the list of pending personal dictionary additions.
+    """
+    try:
+        bot.memory['spellcheck_pending_adds'].remove(trigger.group(2))
+    except ValueError:
+        bot.say('Not in pending list: "{0}"'.format(trigger.group(2)))
+    else:
+        remaining = len(bot.memory['spellcheck_pending_adds'])
+        if remaining == 0:
+            remaining = 'No words'
+        elif remaining == 1:
+            remaining = 'One word'
+        else:
+            remaining = '{0} words'.format(remaining)
+        bot.say('Removed "{0}". ({1} remaining in pending list.)'
+                .format(trigger.group(2), remaining))
+
+
+@commands('scclear')
+@require_admin('Only admins may clear the pending word list.')
+def clear_command(bot, trigger):
+    """
+    Clear the list of words pending addition to the bot's personal dictionary.
+    """
+    count = len(bot.memory['spellcheck_pending_adds'])
+    bot.memory['spellcheck_pending_adds'].clear()
```

Comment:
`.clear()` is not a valid method for `list`s in Python 2, so this, and the one in `save_command()` will need to be changed. 

Something like `del bot.memory['spellcheck_pending_adds'][:]` or `bot.memory['spellcheck_pending_adds'][:] = []` should suffice if you want to maintain the same list object (as opposed to replacing it with a new `[]`).

Reply:
Bleh, the documentation I read didn't mention when `.clear()` was added to lists. I'll fix that (and below).